lib/logger/lib/logger/watcher.ex

```diff
@@ -28,18 +26,24 @@ defmodule Logger.Watcher do
   Removes the given handler.
   """
   def unwatch(mod, handler) do
-    GenEvent.remove_handler(mod, handler, :ok)
+    result = :gen_event.delete_handler(mod, handler, :ok)
+    _ = Supervisor.terminate_child(@name, {__MODULE__, {mod, handler}})
+    result
```

Comment:
Sorry, I'm not sure I understand what you're suggesting. Could you share some code so I can understand what alternative you are proposing?